cleanlab_studio/studio/trustworthy_language_model.py

```diff
@@ -72,119 +69,97 @@ def __init__(
         self,
         api_key: str,
         quality_preset: TLMQualityPreset,
-        max_concurrent_requests: int = _DEFAULT_MAX_CONCURRENT_TLM_REQUESTS,
+        *,
+        options: Optional[TLMOptions] = None,
+        timeout: Optional[float] = None,
+        verbose: Optional[bool] = None,
     ) -> None:
         """Initializes TLM interface.
 
         Args:
             api_key (str): API key used to authenticate TLM client
             quality_preset (TLMQualityPreset): quality preset to use for TLM queries
-            max_concurrent_requests (int): maximum number of concurrent requests when issuing batch queries. Default is 16.
+            options (None | TLMOptions, optional): dictionary of options to pass to prompt method, defaults to None
+            timeout (float, optional): timeout (in seconds) to run all prompts, defaults to None
+            verbose (bool, optional): verbosity level for TLM queries, default to True which will print progress bars for TLM queries. For silent TLM progress, set to False.
         """
         self._api_key = api_key
 
-        assert (
-            max_concurrent_requests < _MAX_CONCURRENT_TLM_REQUESTS_LIMIT
-        ), f"max_concurrent_requests must be less than {_MAX_CONCURRENT_TLM_REQUESTS_LIMIT}"
-
         if quality_preset not in _VALID_TLM_QUALITY_PRESETS:
             raise ValueError(
                 f"Invalid quality preset {quality_preset} -- must be one of {_VALID_TLM_QUALITY_PRESETS}"
             )
 
         self._quality_preset = quality_preset
 
-        if is_notebook():
+        # TODO: validate options args at initialization?
+        if not (options is None or isinstance(options, dict)):
+            raise ValueError(
+                "options must be a TLMOptions object.\n"
+                "See: https://help.cleanlab.ai/reference/python/trustworthy_language_model/#class-tlmoptions"
+            )
+
+        self._options = options
+
+        if timeout is not None:
+            self._timeout = None if timeout <= 0 else timeout
+        else:
+            # TODO: figure out how to compute appropriate timeout
+            self._timeout = 3600
```

Comment:
Per the api spec:

> - `timeout` semantics
>     - If not specified, calculated automatically
>     - If `<= 0`, then infinite
>     - Otherwise: per-prompt timeout, in seconds

I'm not entirely sure what _calculated automatically_ means here, but my understanding was that we would adjust it (based on something)

Reply:
Ok that's bizarre - I've never seen a timeout arg constructed like that.

Can you message Anish and ask for clarification there? I can't think of anything that would be a "reasonable timeout" here